docs/FAQ.rst

```diff
@@ -0,0 +1,273 @@
+security model or enclaves. Docker makes it easier to build and setup Avalon,
+but you can also build without Docker
+(although it requires several more steps).
+
+
+TEEs and Intel® SGX
+===================
+
+What is a Trusted Execution Environment?
+----------------------------------------
+A Trusted Execution Environment (TEE) is a secure area of a processor.
+It ensures code and data is kept secure from the outside environment
+and maintains integrity of input and output with authentication.
+
+A TEE can be thought of as a "reverse sandbox". A traditional sandbox
+restricts the software from accessing system or external resources on a
+machine (such as a Java VM). A TEE "reverse sandbox" or enclave keeps the
+system--other applications and even the OS kernel--from
+accessing data inside the enclave.
+Intel SGX is a hardware implementation of a TEE and is supported by Avalon.
+
+Is Intel SGX required to use Avalon?
+------------------------------------
+No. You can use the Intel SGX simulator to simulate a TEE.
+In the future we plan to add other trusted workers such as
+other hardware TEEs, MPC (multi-party compute), and
+ZK (zero-knowledge proofs).
+
+What is the working principle of Intel SGX TEE Workers?
+-------------------------------------------------------
+At high level you design an application so the core business part resides in
+the enclave, ensuring that even if your untrusted part is compromised the
+trusted part cannot be.
+
+Intel SGX guarantees code and data is kept private and that the results are
+correct with confidentiality and integrity mechanisms.
+
+The PDF link at this webpage gives a good technical overview of Intel SGX enclaves:
+https://software.intel.com/en-us/blogs/2016/06/06/overview-of-intel-software-guard-extension-enclave
+
+
+How can I create a TEE with Intel SGX using Avalon?
+----------------------------------------------------------------------------------
+Start with the examples and tutorial at https://github.com/hyperledger/avalon/tree/master/docs#tutorial
+The technical details of Intel SGX enclaves are encapsulated in the Avalon libraries and Avalon Enclave Manager.
+If you want to learn about low-level details, I would look at the Intel SGX SDK and example programs.
+
+
+I get the message ``intel_sgx: SGX is not enabled`` in ``/var/log/syslog``
+--------------------------------------------------------------------------
+Intel SGX needs to be enabled in BIOS.
+
+Is there a maximum size of input data when using Intel SGX?
+-----------------------------------------------------------
+Avalon doesn’t expect application-specific code to use SGX sealed data.
+Avalon uses sealed data internally for storing private enclave signing and
+encryption keys.
+As result application specific data size is not dependent on the sealed data
+storage.
+It is indirectly limited by the maximum enclave size
+(enclave includes both data and code).
+The maximum enclave size is limited to 128Mbytes.
```

Comment:
EPC size depends on the platform.  Fore example, Intel Xeon E Mehlow-R has 256MB EPC, Xeon E3 is 128MB.

Reply:
I do not want to mention possible future products here. I'll add Menlow.

Reply:
I dropped the changes for this PR.  See PR #242 for changes.